zb-db/src/test/java/io/camunda/zeebe/db/impl/ColumnFamilyPropertyTest.java

```diff
@@ -0,0 +1,205 @@
+ */
+package io.camunda.zeebe.db.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.db.ZeebeDbFactory;
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.jqwik.api.Arbitraries;
+import net.jqwik.api.Combinators;
+import net.jqwik.api.ForAll;
+import net.jqwik.api.Property;
+import net.jqwik.api.Provide;
+import net.jqwik.api.arbitraries.ListArbitrary;
+import net.jqwik.api.lifecycle.AfterTry;
+import net.jqwik.api.lifecycle.BeforeProperty;
+import org.assertj.core.util.Files;
+
+public class ColumnFamilyPropertyTest {
+
+  private Map<Long, Long> map;
+  private ColumnFamily<DbLong, DbLong> columnFamily;
+
+  @Property
+  void columnFamilyBehavesLikeMap(
+      @ForAll("operations") final Iterable<TestableOperation> operations) {
+    operations.forEach(op -> op.runOnMap(map));
+    operations.forEach(op -> op.runOnColumnFamily(columnFamily));
+    columnFamilyBehavedLikeMap(columnFamily, map);
+  }
+
+  private void columnFamilyBehavedLikeMap(
+      final ColumnFamily<DbLong, DbLong> columnFamily, final Map<Long, Long> map) {
+    map.forEach(
+        (key, value) -> {
+          final var dbKey = new DbLong();
+          dbKey.wrapLong(key);
+          assertThat(columnFamily.get(dbKey).getValue())
+              .as("Key " + dbKey.getValue() + " should have value " + value)
+              .isEqualTo(value);
+        });
+    columnFamily.forEach(
+        (key, value) -> assertThat(map).containsEntry(key.getValue(), value.getValue()));
+  }
+
+  @Provide
+  ListArbitrary<TestableOperation> operations() {
+    final var k = Arbitraries.longs().greaterOrEqual(0);
+    final var v = Arbitraries.longs();
+    final var insert =
+        Combinators.combine(k, v)
+            .as(
+                (arbitraryK, arbitraryV) ->
+                    (TestableOperation) new InsertOp(arbitraryK, arbitraryV));
+    final var update =
+        Combinators.combine(k, v)
+            .as(
+                (arbitraryK, arbitraryV) ->
+                    (TestableOperation) new UpdateOp(arbitraryK, arbitraryV));
+    final var upsert =
+        Combinators.combine(k, v)
+            .as(
+                (arbitraryK, arbitraryV) ->
+                    (TestableOperation) new UpsertOp(arbitraryK, arbitraryV));
+    final var deleteExisting =
+        Combinators.combine(k, v)
+            .as(
+                (arbitraryK, arbitraryV) ->
+                    (TestableOperation) new DeleteExistingOp(arbitraryK, arbitraryV));
+    final var c =
+        Combinators.combine(k, v)
+            .as(
+                (arbitraryK, arbitraryV) ->
+                    (TestableOperation) new DeleteIfExistsOp(arbitraryK, arbitraryV));
+
+    return Arbitraries.oneOf(List.of(insert, update, upsert, deleteExisting)).list();
+  }
+
+  @BeforeProperty
+  private void setup() {
```

Comment:
:wrench: Personally, I prefer to put these methods in the order
* setup()
* property()
* cleanup()
* operations()
* helper classes